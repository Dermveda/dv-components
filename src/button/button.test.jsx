import React from 'react';
import { shallow } from 'enzyme';
import Button from './button.jsx';
import { Link } from 'react-router-dom';
import sinon from 'sinon';
const props = {
	type: 'primary',
};
//see API DOCS here: http://airbnb.io/enzyme/docs/api/shallow.html
describe('<Button />', () => {
	it('Will render', () => {
		const wrapper = shallow(<Button {...props} />);
		expect(wrapper).toMatchSnapshot();
	});
	it('Will be secondary button', () => {
		const localProps = { ...props, type: 'secondary' }
		const wrapper = shallow(<Button {...localProps} />);
		expect(wrapper).toMatchSnapshot();
	});
	it('Will be loading', () => {
		const localProps = { ...props, isLoading: true }
		const wrapper = shallow(<Button {...localProps} />);
		expect(wrapper).toMatchSnapshot();
	});
	it('Will show text', () => {
		const localProps = { ...props, text: 'Hey there!' }
		const wrapper = shallow(<Button {...localProps} />);
		expect(wrapper).toMatchSnapshot();
	});
	it('Will be link', () => {
		const localProps = { ...props, to: '/home', isLink: true }
		const wrapper = shallow(<Button {...localProps} />);
		expect(wrapper).toMatchSnapshot();
	});
	it('Will have new target', () => {
		const localProps = { ...props, target: '__blank' }
		const wrapper = shallow(<Button {...localProps} />);
		expect(wrapper).toMatchSnapshot();
	});
	it('Will be submit button', () => {
		const localProps = { ...props, isSubmit: true }
		const wrapper = shallow(<Button {...localProps} />);
		expect(wrapper.find('button').prop('type')).toEqual('submit');
	});
	it('Simulates click', () => {
		const onButtonClick = sinon.spy();
		const wrapper = shallow(<Button onClick={onButtonClick} />);
		wrapper.find('button').simulate('click');
		expect(onButtonClick.calledOnce).toEqual(true);
	});
});