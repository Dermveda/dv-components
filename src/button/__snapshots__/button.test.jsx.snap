// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Will be link 1`] = `
<Link
  className="btn dvc-button undefined 
		dvc-button--primary
		dvc-primary-background
		 
		 
		
		 
		"
  replace={false}
  to="/home"
  type=""
>
  <div
    className="d-flex justify-content-center align-items-center"
  >
    <i
      className="dvc-button__loader"
      key="loader_comp"
    />
  </div>
</Link>
`;

exports[`<Button /> Will be link deep 1`] = `
<a
  className="btn dvc-button undefined 
		dvc-button--primary
		dvc-primary-background
		 
		 
		
		 
		"
  disabled={undefined}
  href="/home"
  onClick={[Function]}
  target={undefined}
  type=""
>
  <div
    className="d-flex justify-content-center align-items-center"
  >
    <i
      className="dvc-button__loader"
    />
    
    
  </div>
</a>
`;

exports[`<Button /> Will be loading 1`] = `
<button
  className="btn dvc-button undefined 
		dvc-button--primary
		dvc-primary-background
		 
		 
		
		 
		"
  disabled={true}
  type="button"
>
  <div
    className="d-flex justify-content-center align-items-center"
  >
    <i
      className="dvc-button__loader mr-2 dvc-button__loader--show"
      key="loader_comp"
    />
  </div>
</button>
`;

exports[`<Button /> Will be loading deep 1`] = `
<button
  className="btn dvc-button undefined 
		dvc-button--primary
		dvc-primary-background
		 
		 
		
		 
		"
  disabled={true}
  onClick={undefined}
  type="button"
>
  <div
    className="d-flex justify-content-center align-items-center"
  >
    <i
      className="dvc-button__loader mr-2 dvc-button__loader--show"
    />
    
    
  </div>
</button>
`;

exports[`<Button /> Will be secondary button 1`] = `
<button
  className="btn dvc-button undefined 
		
		dvc-primary-background
		 
		 
		
		 
		"
  type="button"
>
  <div
    className="d-flex justify-content-center align-items-center"
  >
    <i
      className="dvc-button__loader"
      key="loader_comp"
    />
  </div>
</button>
`;

exports[`<Button /> Will have new target 1`] = `
<button
  className="btn dvc-button undefined 
		dvc-button--primary
		dvc-primary-background
		 
		 
		
		 
		"
  type="button"
>
  <div
    className="d-flex justify-content-center align-items-center"
  >
    <i
      className="dvc-button__loader"
      key="loader_comp"
    />
  </div>
</button>
`;

exports[`<Button /> Will have new target deep 1`] = `
<button
  className="btn dvc-button undefined 
		dvc-button--primary
		dvc-primary-background
		 
		 
		
		 
		"
  disabled={undefined}
  onClick={undefined}
  type="button"
>
  <div
    className="d-flex justify-content-center align-items-center"
  >
    <i
      className="dvc-button__loader"
    />
    
    
  </div>
</button>
`;

exports[`<Button /> Will render 1`] = `
<button
  className="btn dvc-button undefined 
		dvc-button--primary
		dvc-primary-background
		 
		 
		
		 
		"
  type="button"
>
  <div
    className="d-flex justify-content-center align-items-center"
  >
    <i
      className="dvc-button__loader"
      key="loader_comp"
    />
  </div>
</button>
`;

exports[`<Button /> Will render deep 1`] = `
<button
  className="btn dvc-button undefined 
		dvc-button--primary
		dvc-primary-background
		 
		 
		
		 
		"
  disabled={undefined}
  onClick={undefined}
  type="button"
>
  <div
    className="d-flex justify-content-center align-items-center"
  >
    <i
      className="dvc-button__loader"
    />
    
    
  </div>
</button>
`;

exports[`<Button /> Will show text 1`] = `
<button
  className="btn dvc-button undefined 
		dvc-button--primary
		dvc-primary-background
		 
		 
		
		 
		"
  type="button"
>
  <div
    className="d-flex justify-content-center align-items-center"
  >
    <i
      className="dvc-button__loader"
      key="loader_comp"
    />
    <span
      key="text_comp"
    >
      Hey there!
    </span>
  </div>
</button>
`;

exports[`<Button /> Will show text deep 1`] = `
<button
  className="btn dvc-button undefined 
		dvc-button--primary
		dvc-primary-background
		 
		 
		
		 
		"
  disabled={undefined}
  onClick={undefined}
  type="button"
>
  <div
    className="d-flex justify-content-center align-items-center"
  >
    <i
      className="dvc-button__loader"
    />
    
    <span>
      Hey there!
    </span>
  </div>
</button>
`;

exports[`<Button /> Willbe secondary button deep 1`] = `
<button
  className="btn dvc-button undefined 
		
		dvc-primary-background
		 
		 
		
		 
		"
  disabled={undefined}
  onClick={undefined}
  type="button"
>
  <div
    className="d-flex justify-content-center align-items-center"
  >
    <i
      className="dvc-button__loader"
    />
    
    
  </div>
</button>
`;
